core/src/test/java/nl/nn/adapterframework/http/cxf/SoapProviderTest.java

```diff
@@ -318,4 +312,54 @@ public void receiveAndSendMessageWithAttachmentsTest() throws Throwable {
 		//Validate the listener returned an attachment back
 		assertAttachmentInReceivedMessage(message);
 	}
+
+	@Test
+	public void soapActionInSessionKeySOAP1_1() throws Throwable {
+		// Soap protocol 1.1 
+		SOAPMessage request = createMessage("soapmsg1_1.xml", false, true);
+		String value = "1.1-SoapAction";
+		webServiceContext.getMessageContext().put("SOAPAction", value);
+		SOAPProvider.invoke(request);
+		webServiceContext.getMessageContext().clear();
+		assertEquals(value, SOAPProvider.getSession().get("SOAPAction"));
+	}
+
+	@Test
+	public void soapActionInSessionKeySOAP1_2ActionIsTheLastItem() throws Throwable {
+		SOAPMessage request = createMessage("soapmsg1_2.xml");
+		String value = "SOAP1_2ActionIsTheLastItem";
+		webServiceContext.getMessageContext().put("Content-Type", "application/soap+xml; action="+value);
+		SOAPProvider.invoke(request);
+		webServiceContext.getMessageContext().clear();
+		assertEquals(value, SOAPProvider.getSession().get("SOAPAction"));
+	}
+
+	@Test
+	public void soapActionInSessionKeySOAP1_2ActionIsInMiddle() throws Throwable {
+		SOAPMessage request = createMessage("soapmsg1_2.xml");
+		String value = "SOAP1_2ActionIsInMiddle";
+		webServiceContext.getMessageContext().put("Content-Type", "application/soap+xml; action="+value+";somethingelse");
+		SOAPProvider.invoke(request);
+		webServiceContext.getMessageContext().clear();
+		assertEquals(value, SOAPProvider.getSession().get("SOAPAction"));
+	}
+
+	@Test
+	public void soapActionInSessionKeySOAP1_2ActionIsAtTheBeginning() throws Throwable {
+		SOAPMessage request = createMessage("soapmsg1_2.xml");
+		String value = "SOAP1_2ActionIsAtTheBeginning";
+		webServiceContext.getMessageContext().put("Content-Type", "action="+value+";application/soap+xml; somethingelse");
+		SOAPProvider.invoke(request);
+		webServiceContext.getMessageContext().clear();
+		assertEquals(value, SOAPProvider.getSession().get("SOAPAction"));
+	}
+
+	@Test
+	public void noSoapActionInSessionKey() throws Throwable {
+		SOAPMessage request = createMessage("soapmsg1_2.xml");
+		webServiceContext.getMessageContext().put("Content-Type", "application/soap+xml; somethingelse");
+		SOAPProvider.invoke(request);
+		webServiceContext.getMessageContext().clear();
+		assertNull(SOAPProvider.getSession().get("SOAPAction"));
+	}
```

Comment:
All these tests could also be simplified by just testing `findAction(contentType)` with all these different contentTypes.
I would much rather see a test with soap1.1 and a soap 1.2 header, and vice-versa.
Happy flow vs unhappy flow